### PR TITLE
MIDI-123: Documentation and slight changes

### PR DESCRIPTION
**Pull request by SamuelJanas**
[MIDI-123](https://sacrebleu.atlassian.net/browse/MIDI-123)

[MIDI-123]: https://sacrebleu.atlassian.net/browse/MIDI-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by SamuelJanas** [comment]
Expected those test to fail but wanted to push this change here to describe issues for future development.
I've came up with a nice way of testing the apply_sustain function, however there is an issue when using equals on pandas dfs, mainly with float precision unfortunately. I'll be trying to fix it with [numpy.isclose](https://numpy.org/doc/stable/reference/generated/numpy.isclose.html) or some other method.

**Comment by SamuelJanas** [comment]
Seems like there is still some issue in our new theme. Following the official [tutorial](https://squidfunk.github.io/mkdocs-material/customization/) didn't help too much. The extra_css argument doesn't seem to be working at all, it might be the issue with pallete and toggling. For now the color works by specifying the value in mkdocs.yaml. I'll try to look for an answer as some people also had [similar issues](https://stackoverflow.com/questions/75231063/including-extra-css-with-mkdocs)

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
-# fortepyan :musical_keyboard:
+# Fortepyan :musical_keyboard:
+![GitHub CI](https://github.com/Nospoko/Fortepyan/actions/workflows/ci_tests.yaml/badge.svg?branch=master) [![PyPI version](https://img.shields.io/pypi/v/fortepyan.svg)](https://pypi.org/project/fortepyan/) [![PyPI download month](https://img.shields.io/pypi/dm/fortepyan.svg)](https://pypi.org/project/fortepyan/)
```
Comment (SamuelJanas) [inline README.md]: @roszcz Let me know If you'd like to see some other badges as well!
image

**Comment by roszcz** [comment]
@SamuelJanas this one looks cool: https://shields.io/badges/py-pi-python-version

Right now it generates this:

PyPI - Python Version

I'm not sure what would we need to change to get this:

image

probably this would require some changes in package setup and build process 🤔

**Comment by SamuelJanas** [comment]
> @SamuelJanas this one looks cool: https://shields.io/badges/py-pi-python-version
> 
> Right now it generates this:
> 
> PyPI - Python Version
> 
> I'm not sure what would we need to change to get this:
> 
> image
> probably this would require some changes in package setup and build process 🤔

A static one will do for that. 
```
[![Python 3.9](https://img.shields.io/badge/python-3.9+-blue.svg)](https://www.python.org/downloads)
```

[Python 3.9](https://www.python.org/downloads)

If you want a dynamic version I think it's a matter of adding 
```
        'Programming Language :: Python :: 3.9',
        'Programming Language :: Python :: 3.10',
        'Programming Language :: Python :: 3.11',
        'Programming Language :: Python :: 3.12',
```
to classifiers in `pyproject.toml`. LMK what works best for you

**Comment by roszcz** [comment]
I think 3.9+ is great :+1:

**Comment by SamuelJanas** [comment]
From what I understand, the new workflow should have created a page at `<username>.github.io/<repository>` which would be [Nospoko.github.io/fortepyan](https://github.com/Nospoko/fortepyan) but the redirect to algrtm may be affecting it somehow? I'll have to look into that unless you have any idea what's happening. 😄  

EDIT: works.

**Comment by roszcz** [comment]
@SamuelJanas I updated the branch in repo settings/pages and everything seems alive: https://algrtm.com/fortepyan/

File: docs/assets/extra.css [hunk 2]
```diff
@@ -1,3 +1,9 @@
 :root {
-    --md-primary-fg-color: #1c394a;
+    --md-primary-fg-color: #944038;
   }
+
+[data-md-color-scheme="slate"] {
+  --md-hue: 210;
+  /* for better contrast make the links bright */
+  --md-typeset-a-color: #E8A03E;
```
Comment (SamuelJanas) [inline docs/assets/extra.css]: It works very selectively I'd say. for now I'm [linking](https://github.com/squidfunk/mkdocs-material/blob/6b5035f5580f97532d664e3d1babf5f320e88ee9/src/assets/stylesheets/main/_colors.scss) their css colors template for possible future changes.

This issue https://github.com/squidfunk/mkdocs-material/issues/2987 might be helpful as well

**Comment by roszcz** [comment]
Is it possible to have more granular control over what is displayed in the documentation? I'm not sure if it's necessary to show descriptions of functions that could be considered "private". `__sanitize_get_index` is on example. Similarly I'm not sure why is there a `soundfont` section in the render docs:

image

Seems to be empty - can we do something to not display it? :)

**Comment by SamuelJanas** [comment]
> Is it possible to have more granular control over what is displayed in the documentation? I'm not sure if it's necessary to show descriptions of functions that could be considered "private". `__sanitize_get_index` is on example. Similarly I'm not sure why is there a `soundfont` section in the render docs:
> 
> image
> Seems to be empty - can we do something to not display it? :)

Ah, sorry! I must have missed the notification. 
From what I understand:
```python
These functions provide a simple interface to render audio files from MIDI representations.

::: fortepyan.audio.render

::: fortepyan.audio.soundfont
```
This is all the code for the `rendering` docs, both section render and soundfont are pointing to the beginning of documentation for that module. If we were to write some module docstrings inside soundfont, clicking on it would direct to that docstring.

We can hide it using 
```
::: fortepyan.audio.soundfont
    options:
        show_root_toc_entry: false
```
or setting this option globally.

EDIT: Regerding the private methods, we can add filters to our options.
```yaml
  - mkdocstrings:
      handlers:
        python:
          options:
            filters:
              - "^(?!__).*$"
```
This would filter out anything that starts with double underscore

**Review by roszcz — APPROVED** [review]

**Pull request merged**